mono/metadata/class.c

```diff
@@ -760,18 +831,82 @@ inflate_generic_type (MonoImage *image, MonoType *type, MonoGenericContext *cont
 	}
 	case MONO_TYPE_PTR: {
 		MonoType *nt, *inflated = inflate_generic_type (image, type->data.type, context, error);
-		if (!inflated || !mono_error_ok (error))
+		if ((!inflated && !changed) || !mono_error_ok (error))
 			return NULL;
+		if (!inflated && changed)
+			return type;
 		nt = mono_metadata_type_dup (image, type);
 		nt->data.type = inflated;
 		return nt;
 	}
 	default:
-		return NULL;
+		if (!changed)
+			return NULL;
+		else
+			return type;
 	}
 	return NULL;
 }
 
+static MonoType*
+inflate_generic_custom_modifiers (MonoImage *image, const MonoType *type, MonoGenericContext *context, MonoError *error)
+{
+	MonoType *result = NULL;
+	g_assert (type->has_cmods);
+	int count = mono_type_custom_modifier_count (type);
+	gboolean changed = FALSE;
+	MonoType *new_type = g_alloca (mono_sizeof_type_with_mods (count, TRUE));
+	memcpy (new_type, type, MONO_SIZEOF_TYPE);
```

Comment:
`*new_type = *type`?

Reply:
Previously it needed to be an alloc + some copies because mono_types with cmods are bigger than sizeof(type)

Reply:
yea this line probably merits a comment or a helper function - I am specifically just copying the "plain type" slice of `type` into `new_type`.  It's true that `*new_type = *type` would do the same thing, but it's perhaps even less evident than a memcpy.

Reply:
It should be obvious, at least in C.